spec/valvat/lookup_spec.rb

```diff
@@ -240,4 +304,132 @@
       end
     end
   end
+
+  describe '#validate with HMRC test enviroment' do
+    # https://developer.service.hmrc.gov.uk/api-documentation/docs/testing
+    # https://github.com/hmrc/vat-registered-companies-api/blob/master/public/api/conf/1.0/test-data/vrn.csv
+    before do
+      stub_const('Valvat::Lookup::HMRC::Request::HMRC_API_URL', 'https://test-api.service.hmrc.gov.uk/organisations/vat/check-vat-number/lookup')
+    end
+
+    subject(:result) { described_class.validate('GB123456789') }
+
+    context 'with valid request with valid VAT number' do
+      it 'returns true' do
+        expect(described_class.validate('GB553557881')).to be(true)
+        expect(described_class.validate('GB146295999727')).to be(true)
+      end
+    end
+
+    context 'with valid request with an invalid VAT number' do
+      it 'returns false' do
+        expect(result).to be(false)
+      end
+    end
+
+    describe 'Error : MESSAGE_THROTTLED_OUT' do
+      before {
+        stub_request(:get, /test-api.service.hmrc.gov.uk/).to_return(
+          status: 429,
+          body: '{"code":"MESSAGE_THROTTLED_OUT"}'
+        )
```

Comment:
I had to add `webmock` to the dev dependencies to be able to test error responses as their test API do not offer some special VAT number to generate errors, they only offer different fake but valid numbers.

Reply:
Ok sorry again I found them in https://github.com/yolk/valvat/blob/9266fd143fea6eefb96ece9cde1f2efedcbbc887/spec/valvat/lookup_spec.rb :)
resolving this too :)

Reply:
No intention per se ... yepp, seems to be a good idea to test against these stubbed error cases. Thx!

Reply:
Parallel again :-) ... okay, I'll need to practice myself in some patience.

Reply:
Mee too xD